servers/v3-proxy-api/src/graph/shared/transforms.ts

```diff
@@ -0,0 +1,189 @@
+/**
+ * method to convert graph responses to REST responses
+ */
+
+import {
+  Tag,
+  Author,
+  DomainMetadata,
+  Image,
+  Video,
+  ItemCompleteFragment,
+  Imageness,
+  Videoness,
+} from '../../generated/graphql/types';
+import {
+  TagsItemObject,
+  AuthorsItemObject,
+  DomainMetadataItemObject,
+  ImagesItemObject,
+  VideosItemObject,
+  VideoTypeMap,
+  AddAuthorsObject,
+} from '../types';
+
+/**
+ * Convert tag response array into a map keyed by tag name
+ */
+export function TagsReducer(
+  tags: Array<Pick<Tag, 'name'>> | undefined,
+  itemId: string,
+): TagsItemObject | undefined {
+  if (tags == null) {
+    return undefined;
+  }
+  return tags.reduce((tagsObj, tag) => {
+    tagsObj[tag.name] = {
+      tag: tag.name,
+      item_id: itemId,
+    };
+    return tagsObj;
+  }, {} as TagsItemObject);
+}
+
+/**
+ * Convert authors response array into a map keyed by author id
+ */
+export function AuthorsReducer(
```

Comment:
These overloads were added because the shape differs for /v3/get vs. /v3/add (does and does *not* include item_id key, respectively)